internal/fnapi/naming.go

```diff
@@ -83,14 +83,22 @@ func AllocateName(ctx context.Context, opts AllocateOpts) (*NameResource, error)
 
 		var nr IssueResponse
 
+		fetchTenantToken := func(ctx context.Context) (string, error) {
+			t, err := FetchTenantToken(ctx)
+			if err != nil {
+				return "", err
+			}
+			return t.Raw(), nil
+		}
+
 		if err := (Call[IssueRequest]{
 			Endpoint: EndpointAddress,
 			Method:   "nsl.naming.NamingService/Issue",
 			PreAuthenticateRequest: func(ua *auth.UserAuth, rt *IssueRequest) error {
 				rt.UserAuth = ua
 				return nil
 			},
-			FetchToken: auth.GenerateToken,
+			FetchToken: fetchTenantToken,
```

Comment:
We also need to update https://github.com/namespacelabs/foundation/blob/0ba4493266b26d72c9a9f2959824e347189139bd/internal/fnapi/map.go#L35-L36